donate/payments/views.py

```diff
@@ -0,0 +1,246 @@
+                default_error_message = _('Sorry there was an error processing your payment. '
+                                          'Please try again.')
+                form.add_error(None, default_error_message)
+
+            return self.form_invalid(form)
+
+    def get_custom_fields(self, form):
+        return {}
+
+    def get_merchant_account_id(self):
+        return settings.BRAINTREE_MERCHANT_ID
+
+    def filter_user_card_errors(self, result):
+        client_errors = {
+            ErrorCodes.CreditCard.CreditCardTypeIsNotAccepted: _('The type of card you used is not accepted.'),
+            ErrorCodes.CreditCard.CvvIsInvalid: _('The CVV code you entered was invalid.'),
+            ErrorCodes.CreditCard.CvvVerificationFailed: _('The CVV code you entered was invalid.'),
+            ErrorCodes.CreditCard.ExpirationDateIsInvalid: _('The expiration date you entered was invalid.'),
+            ErrorCodes.CreditCard.NumberIsInvalid: _('The credit card number you entered was invalid.'),
+        }
+        return [
+            client_errors[error.code] for error in result.errors.deep_errors
+            if error.code in client_errors.keys()
+        ]
+
+    def check_for_address_errors(self, result):
+        errors = {
+            ErrorCodes.Address.PostalCodeInvalidCharacters: _('The post code you provided is not valid.'),
+            ErrorCodes.Address.PostalCodeIsTooLong: _('The post code you provided is not valid.'),
+        }
+        for error in result.errors.deep_errors:
+            if error.code in errors:
+                # The view is expected to catch this exception and pass the errors
+                # back to the personal details view so that the use can try to correct them.
+                raise InvalidAddress(errors=[errors[error.code]])
+
+    def success(self, result, form):
+        return HttpResponseRedirect(self.get_success_url())
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        ctx.update({
+            'braintree_tokenization_key': settings.BRAINTREE_TOKENIZATION_KEY,
+            'payment_amount': self.personal_details['amount'],
+            'braintree_params': {
+                'use_sandbox': settings.BRAINTREE_USE_SANDBOX,
+            }
+        })
+        return ctx
+
+
+class PersonalDetailsView(FormView):
+    form_class = PersonalDetailsForm
+    template_name = 'payment/personal_details.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        return super().dispatch(request, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        # Ensure that the donation amount is a valid currency decimal
+        # (either only a whole number, or a number with up to 2 decimal places)
+        if not re.match(r'^\d+(?:\.\d{1,2})?$', self.request.GET.get('amount', '50')):
+            return HttpResponseRedirect(self.campaign_source_page.get_url(request))
```

Comment:
Note to self - `campaign_source_page` isn't implemented yet and I need to remove this until it is.